### DISCUSSION TITLE
Cannot grab models from HuggingFace or from local GGUF files

### DISCUSSION THREAD

**Discussion by rchaklashiya** [open-webui/open-webui#7998] [Issues]
## Installation Method

Docker Engine on Linux via "sudo ollama serve" followed by "`sudo docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama`

## Environment

- **Open WebUI Version:** 0.4.8
- **Ollama (if applicable):** 0.5.1

- **Operating System:** Ubuntu 24.04.1 LTS
- **Browser (if applicable):** Firefox

## Expected Behavior:

Based on what I saw in the docs (see here: https://docs.openwebui.com/features/#-model-management ) I should be able to select huggingface models or local GGUF files from going to a "experimental" section in the models tab under the admin settings. Specifically, the documentation says this:
"⬆️ GGUF File Model Creation: Effortlessly create Ollama models by uploading GGUF files directly from Open WebUI from the Admin Settings > Settings > Model > Experimental menu. The process has been streamlined with the option to upload from your machine or download GGUF files from Hugging Face."

## Actual Behavior:
There is no experimental menu, nor does simply doing something like putting the huggingface model url into the models searchbar to pull the model working. For the model searchbar option, I chose the example: huggingface.co/chaoticneutrals/tensorblock/hathor_respawn-l3-8b-v0.8-gguf:q4_k_m and trying to pull it doesn't work, see screenshot (gives the "download cancelled" error). I do not see any way to grab a local GGUF file.

## Logs and Screenshots
![image](https://github.com/user-attachments/assets/3c0b824d-d341-4e09-9327-7e5982da9597)
![image](https://github.com/user-attachments/assets/cc95cf8e-65b2-4faf-a30f-17cdd1453ffe)

**Comment by JohnathanCrow** [comment]
Not sure about pulling direct, but for local: Admin Settings > Connections > Manage Ollama Connections (Spanner Icon) > Experimental (Show)

Some of the recent UI changes have been a little confusing for me too. I'd already found the new location for upload before, but it still took me a minute to remember. I think I read that the docs are kind of in progress, so some things are probably outdated.

**Reply by RichardFevrier** [reply]
Thanks I would never have found this without your guidance. So unintuitive.

**Comment by joequant** [comment]
There is a much easier way.  If you just put in the hf.co in the ollama pull dialog box i.e.

hf.co/DavidAU/Gemma-The-Writer-N-Restless-Quill-10B-Uncensored-GGUF:Q8_0

This will add new hugging face models.

**Reply by edgartaor** [reply]
Nice trick!
If that is already implemented ,maybe it should be an option on the menu. "Pull <model>from HuggingFace"

**Comment by theKHR3b** [comment]
Just small help from me: On hugging.face on model page with gguf just click on the right up button called "**Use this model**". Select in dropdown menu Ollama. In opened window select quantisation what you need, and the press "**copy**" button. Paste this link default pull ollama dialog box, like you pull models from ollama.

**Reply by GeorgiaM-honestly** [reply]
Hello! I know it has been a while since this post, and maybe the huggingface page has changed. I see the "Use this model" button, however, I don't see ollama there. Am I missing something?

I am trying to install this:

https://huggingface.co/LyraNovaHeart/Dazzling-Star-Aurora-32b-v0.0-Experimental-1130/tree/main

My Open-WebUI version is the 0.6.31 container with ollama bundled inside of it.

**Reply by silentoplayz** [reply]
@GeorgiaM-honestly The model you linked is not a `GGUF` file. You will see `Ollama` as an option via the "Use this model" button on the HuggingFace page for `GGUF` models. e.g: https://huggingface.co/unsloth/Magistral-Small-2509-GGUF, https://huggingface.co/unsloth/models?search=gguf, https://huggingface.co/models?pipeline_tag=text-generation&library=gguf&sort=trending

**Reply by GeorgiaM-honestly** [reply]
I see. Kind of an obnoxious limitation. How do I install the model I linked to? I've searched around and there is very little information.

**Reply by silentoplayz** [reply]
You can't run the model you linked using Ollama, that's for certain. If there's no GGUF file provided, you must find an alternative method to run the model. HuggingFace provides those other options to you and tells you how to get up and running with those other options. See this HuggingFace blog post for more information. https://huggingface.co/docs/hub/en/ollama. As a bonus, you can read https://huggingface.co/docs/hub/en/gguf

**Reply by mcowger** [reply]
FWIW there is a GGUF quant of that model here:

https://huggingface.co/bartowski/Dazzling-Star-Aurora-32b-v0.0-Experimental-1130-GGUF
